### ISSUE TITLE
Adding a replace attribute function broke feature 

### ISSUE THREAD

**Issue by burgesQ** [lmittmann/tint#37]
I want to add a replace attribute function to handle custom log level & wording (FATAL, TRACE). 

I've noticed that adding a `ReplaceAttrs` functions only handling custom level to string break the source parameters format and level colorization. A weird struct is displayed instead of the colored `file:line` and the levels are greyed out.

Does that mean that I have the re-implement the wheel if I wish to add more human friendly log level ? 

Cheers

**Comment by lmittmann** [comment]
Could you please provide an example test case like https://github.com/lmittmann/tint/blob/main/handler_test.go#L46-L51 so I can better understand what your `ReplaceAttr` function is and what output you expect.

**Comment by burgesQ** [comment]
Sure, the following code : 

<details>
<summary>go code</summary>

```go
package main

import (
	"context"
	"log/slog"
	"os"

	"github.com/lmittmann/tint"
)

func main() {
	slg := slog.New(tint.NewHandler(os.Stderr, &tint.Options{
		Level:      slog.LevelDebug,
		AddSource:  true,
		TimeFormat: "2006/01/02 15:04:05.999999",
		ReplaceAttr: func(g []string, a slog.Attr) slog.Attr {
			if len(g) == 0 && a.Key == slog.LevelKey {
				if a.Value.Int64() == int64(slog.LevelError+4) {
					return slog.String(slog.LevelKey, "FATAL")
				}
			}

			return a
		},
	}))

	slg.Log(context.Background(), slog.LevelDebug, "hello")
	slg.Log(context.Background(), slog.LevelError+4, "hello")
}
```

</details>

Will produce that output (DBG and FATAL are not colored): 

```console
~/repo/test2 
12:59:44 ✘ 1 ❯ go run main.go
2023/09/07 12:59:58.246866 DBG &{main.main /home/master/repo/test2/main.go 27} hello
2023/09/07 12:59:58.246978 FATAL &{main.main /home/master/repo/test2/main.go 28} hello
```

If I omit the ReplaceAttr option, then the output is the classic one: 

```console
~/repo/test2 
12:57:54 ❯ go run main.go
2023/09/07 12:58:33.587326 DBG &{main.main /home/master/repo/test2/main.go 19} hello
2023/09/07 12:58:33.587403 ERR+4 &{main.main /home/master/repo/test2/main.go 20} hello
```



Let me probably rephrase the MR

**Comment by burgesQ** [comment]
Hum and seems the file stripping depends on one some other component - I can produce the following behavior w/o providing any `ReplaceAttr` method: 

![image](https://github.com/lmittmann/tint/assets/21552387/81c66c86-bd69-4323-9565-10db0de93086)

But if I set the following in my project:

<details>
<summary>go code extract </summary>

```go
case OutputTinted:
		lg = slog.New(tint.NewHandler(o.w, &tint.Options{
			Level:      lvl,
			AddSource:  true,
			TimeFormat: "2006/01/02 15:04:05.999999",
			ReplaceAttr: func(groups []string, attr slog.Attr) slog.Attr {
				if len(groups) == 0 && attr.Key == slog.LevelKey {
					if lvl, ok := _iLvl2str[attr.Value.Int64()]; ok {
						return slog.String(slog.LevelKey, lvl)
					}
				}

				return attr
			},
```

</details>

Produce my initial buggy output: 

![image](https://github.com/lmittmann/tint/assets/21552387/aa8d51b8-bd88-4cc9-b073-f201737c4f6d)

**Comment by lmittmann** [comment]
This issue was fixed in #39. Note, that you will need to handle coloring if you replace the default levels.
